gcloud/storage/api.py

```diff
@@ -184,6 +170,11 @@ class _BucketIterator(Iterator):
     """
 
     def __init__(self, connection, extra_params=None):
+        # If an implicit connection was intended, we pass along `None` to the
+        # Bucket() constructor as well.
```

Comment:
Hmm, so I guess the issue is that we can't use the batch to do the iterator's own call, but we want to use it (if it exists) within the buckets returned from the iterator.